daemon/container_operations_unix.go

```diff
@@ -772,6 +768,9 @@ func (daemon *Daemon) connectToNetwork(container *container.Container, idOrName
 	}
 
 	if endpointConfig != nil {
+		if !containertypes.NetworkMode(idOrName).IsUserDefined() && len(endpointConfig.Aliases) > 0 {
+			return runconfig.ErrUnsupportedNetworkAndAlias
+		}
```

Comment:
though it is not required, since these functions already do this check. But am fine making this change.